core/src/main/java/com/spotify/ffwd/output/CoreOutputManager.java

```diff
@@ -241,7 +249,53 @@ private Batch filter(final Batch batch) {
         return mergedTags;
     }
 
-    private String selectHost(Metric metric) {
+    /**
+     * Potentially convert some tags to resource identifiers - i.e. tags in tagsToResource conf.
+     *
+     * If there are conflicts, the existing resource identifiers takes precedence over tags.
+     *
+     * @param tags Map of tags that will be used when constructing a metric
+     * @param resource Map of resource identifiers that will be used when constructing a metric
+     */
+    private void processTagsToResource(
+        final Map<String, String> tags, final Map<String, String> resource
+    ) {
+        if (tagsToResource.isEmpty()) {
+            return;
+        }
+
+        tagsToResource.forEach((fromTag, asResource) -> {
+            // Set as resource if the tag exists and there were not already a resource with the
+            // wanted name.
+            if (tags.containsKey(fromTag) && !resource.containsKey(asResource)) {
+                resource.put(asResource, tags.get(fromTag));
+            }
+            // Always remove the tag
+            tags.remove(fromTag);
+        });
+    }
+
+    private String selectHost(final Metric metric, final Map<String, String> resource) {
```

Comment:
Might be time to consider making host just another tag internally.

Reply:
True. I'll give it a quick stab to see if it's cumbersome or not.

Reply:
I submitted #68, to make host be a tag. I suggest we finalise that one before merging #66.